Add the guide on how to collect guest cluster logs (#724)
harvester/docs: docs/troubleshooting/harvester.md

:::

:::note

> To collect useful logs and configuration files of a guest cluster, please check [Collect Guest Cluster Logs](./rancher.md#collect-guest-cluster-logs).

Reviewer: ```suggestion
For information about collecting guest cluster logs and configuration files, see [Guest Cluster Log Collection](./rancher.md#guest-cluster-log-collection).
```
